### ISSUE TITLE
define "registrable domain" "registered domain" on publicsuffix.org/list

### ISSUE THREAD

**Issue by equalsJeffH** [publicsuffix/publicsuffix.org#12]
the Algorithm section of https://publicsuffix.org/list/ contains this step:
`7.  The registered or registrable domain is the public suffix plus one additional label.`
..but "registered domain" and "registrable domain" are not defined. 

Discussion with `<team@publicsuffix.org>` yielded these definitions:
> 7. The registered (or registrable) domain is the public suffix plus one additional label, where:
> * A registered domain is a domain that is registered, which is a process defined within the domain name system related to IANA functions, ICANN, and domain name registries.
> * A registrable domain is a formally valid domain name that can be registered at a registry, i.e., it is not presently registered. 

Perhaps the above definitions (or refinements thereof) could be added to https://publicsuffix.org/list/ ?   various folks are using the term "registrable domain" in various specs where they likely mean "registered domain" (but the original text of step 7 made it easier to latch onto the term "registrable domain" by default). 

Please see also PR #2.

[this issue supersedes publicsuffix/list#236]

**Comment by annevk** [comment]
See also discussion at https://github.com/publicsuffix/list/issues/27.

**Comment by simon-friedberger** [comment]
These terms are not defined or used precisely. They are also applied to things like somebody.github.io which is not registrable or registered in any sense related to the domain name system, it's just something you can get with an account on GitHub.

The only sensible definition we could make is that a registrable domain or eTLD+1 is a domain on this list with an additional label but that seems kind of circular.

**Comment by mozfreddyb** [comment]
[Registerable domain](https://url.spec.whatwg.org/#host-registrable-domain) is defined in the HTML spec and I think we want these definitions to be identical. I think they are :-)

**Comment by simon-friedberger** [comment]
Yes, but people also use "registered domain" as a domain that is registered with an ICANN-accredited registrar and "registrable domain" for one which could be registered but isn't.

And the HTML spec basically says "registrable domain" is the public suffix from the PSL plus one label so we also cannot use that without having a circular reference.

But indeed `The registered or registrable domain is the public suffix plus one additional label.` from the wiki is actually the definition. Even though such a domain is not necessarily registered or registrable in any colloquial sense of the word.

**Comment by dnsguru** [comment]
YGBFKM was my first reaction here, but, let's burn some volunteer cycles
and define what was abundantly clear more clearly.

"A registerable domain or registered domain are domain names that are
syntactically and technically legal by standards of RFC, structured within
the ICP-3 root namespace or namespace indicated intentionally for inclusion
at their apex by the IANA, IETF, IAB or ICANN.

This includes namespace definitions within descendants of those apexes,
excluding namespace that is not, or new specifications that may create
disruption to existing systems."

We follow ICP-3.  And...
(a) do not want to burn / waste cycles with alt namespaces or web3 etc
technologies.  Many a lost cycle to that energy = stove not to touch
(b) leave room for stuff like .onion that found a side door via IETF vetting
(c) leave room for not including things like .test or .example or .internal
where there may be disruption to long standing norms
(d) leave room for ICANN publishing TLDs to us that are on a trajectory to
be added to the ICP-3 root as TLD applicants contract so as to mitigate our
organic downstream propagation
(e) exclude "illegal" (by technical terms) namespace such as IDN outside of
ranges or namespace exceeding length boundaries.
